src/service-api.ts

```diff
@@ -0,0 +1,299 @@
+          const contentFolder = buildContentPath(targetFolder);
+
+          // save H5P file
+          await pipeline(h5pFile.file, fs.createWriteStream(savePath));
+          await extract(savePath, { dir: contentFolder });
+
+          const result = await h5pValidator.validatePackage(contentFolder);
+          if (!result.isValid) {
+            throw new InvalidH5PFileError(result.error);
+          }
+
+          // try-catch block for remote storage cleanup
+          try {
+            // upload whole folder to public storage
+            await uploadH5P(targetFolder, remoteRootPath, member, log);
+            const item = await createH5PItem(baseName, contentId, member, parentId);
+            return item;
+          } catch (error) {
+            // delete public storage folder of this H5P if upload or creation fails
+            const deleteTask = fileTaskManager.createDeleteFolderTask(member, {
+              folderPath: remoteRootPath,
+            });
+            await taskRunner.runSingle(deleteTask);
+            // rethrow above
+            throw error;
+          }
+          // end of try-catch block for remote storage cleanup
+        } catch (error) {
+          // log and rethrow to let fastify handle the error response
+          log.error('graasp-plugin-h5p: unexpected error occured while importing H5P:');
+          log.error(error);
+          // wrap into plugin error type if not ours
+          if (!(error instanceof GraaspImportH5PError)) {
+            error = new H5PImportError();
+          }
+          throw error;
+        } finally {
+          // in all cases, remove local temp folder
+          await tmpDir.cleanup();
+        }
+        // end of try-catch block for local storage cleanup
+      },
+    );
+
+    /**
+     * Delete H5P assets on item delete
+     */
+    const deleteItemTaskName = itemTaskManager.getDeleteTaskName();
+    taskRunner.setTaskPostHookHandler<Item<H5PExtra>>(deleteItemTaskName, async (item, actor) => {
+      if (item.type !== H5P_ITEM_TYPE) {
+        return;
+      }
+      const deleteTask = fileTaskManager.createDeleteFolderTask(actor, {
+        folderPath: buildRootPath(pathPrefix, item.extra.h5p.contentId),
+      });
+      await taskRunner.runSingle(deleteTask);
+    });
+
+    /**
+     * Copy H5P assets on item copy
+     */
+    const copyItemTaskName = itemTaskManager.getCopyTaskName();
+    taskRunner.setTaskPreHookHandler<Item<H5PExtra>>(copyItemTaskName, async (item, actor) => {
+      // only execute this handler for H5P item types
+      if (item.type !== H5P_ITEM_TYPE) {
+        return;
+      }
+      if (!item.name) {
+        throw new Error('Invalid state: missing previous H5P item name on copy');
+      }
+      if (!item.extra?.h5p) {
+        throw new Error('Invalid state: missing previous H5P item extra on copy');
+      }
+
+      const baseName = path.basename(item.name, H5P_FILE_DOT_EXTENSION);
+      const copySuffix = '-1';
```

Comment:
It's a minus 1? Or to be able to duplicate files in the same directory? Should we go for `Copy` instead?

Reply:
This is used when copying an H5P item, so item `foo` will be copied with name `foo-1` and the physical underying h5p files will be named `foo.h5p` and `foo-1.h5p` respectively. I think we should keep the `-1` suffix instead of `copy` because it is used as the actual file name, it's probably simpler w.r.t i18n and characters that are allowed as file names.